### ISSUE TITLE
Doesn't work when buffer is not modifiable

### ISSUE THREAD

**Issue by kristijanhusak** [kizza/actionmenu.nvim#7]
I tried to set up this with defx, but I constantly get an error that buffer is not modifiable. I tried in regular buffer, same error. Steps to reproduce:

1. Open any file
2. enter `set nomodifiable` in command line
3. call `actionmenu#example()`
4. Select any option
5. See error in command line

I couldn't find what's causing this. Maybe you can.

**Comment by kizza** [comment]
Howdy @kristijanhusak!  Thanks for finding this, and for detailing the steps to reproduce the error. I was indeed able to replicate it myself.

I've just pushed a commit to ensure the actionmenu buffer is always modifiable locally (which means it's able to open the actionmenu successfully).

Hoping this fixes your issue (I haven't used `defx` before).  I'm hoping/expecting that depending on what you're doing with the `callback`, nothing should be trying to modify the buffer now.

**Comment by kristijanhusak** [comment]
@kizza thanks for the quick response. I tried it, and unfortunately it still happens.
Here's a screenshot:
![screenshot](https://i.imgur.com/PzDghO6.png)

Note that i'm using NeoVim latest from master. Maybe it is related to that.

**Comment by kizza** [comment]
Hmmm. When I followed the steps to reproduce I first got
![image](https://user-images.githubusercontent.com/1088717/65552286-5b6ce700-df67-11e9-9ad6-d88b648b7356.png)
So the actionmenu wouldn't even open!  Now it opens (but yes, I get the same error after I select an option, even if the callback doesn't touch the buffer)

I was able to identify that ["closing" the actionmenu window via a different method](https://github.com/kizza/actionmenu.nvim/commit/c3fe06e95570ffe9e28ec75d5c8dc3324f9c1987) solved the problem for me

I can now run through your original steps to reproduce the error, select an item, close the window, run the callback - and all works fine.  Hope you have the same experience too!
